src/nupic/types/Sdr.cpp

```diff
@@ -60,6 +60,14 @@ namespace sdr {
             for(auto idx : sparse_) {
                 NTA_ASSERT(idx < size);
             }
+            NTA_ASSERT( is_sorted(sparse_.begin(), sparse_.end()) )
+                << "Sparse & Coordinate data must be sorted!";
+            UInt previous = -1;
+            for( const UInt idx : sparse_ ) {
+                NTA_ASSERT( idx != previous )
+                    << "Sparse & Coordinate data must not contain duplicates!";
+                previous = idx;
+            }
```

Comment:
nit. The check for is_sorted and the check for duplicates could be done in a single pass rather than two.

Reply:
I took another look at these assert statements.

I did not do as you suggested because I want to have different error messages for sorted & unique.

I remove a different for-loop through the sparse data, by taking advantage of the new sorted property of the data.  I also improved the assert statements for the coordinate format data.

Reply:
Ok, that seems reasonable.